casper/src/test/scala/coop/rchain/casper/MultiParentCasperRemainingTest.scala

```diff
@@ -0,0 +1,400 @@
+package coop.rchain.casper
+
+import java.nio.file.Files
+
+import cats.{Applicative, Monad}
+import cats.data.EitherT
+import cats.effect.{Concurrent, Sync}
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.BlockStore
+import coop.rchain.casper.Estimator.Validator
+import coop.rchain.casper.genesis.Genesis
+import coop.rchain.casper.genesis.contracts._
+import coop.rchain.casper.MultiParentCasper.ignoreDoppelgangerCheck
+import coop.rchain.casper.helper.HashSetCasperTestNode.Effect
+import coop.rchain.casper.helper.{BlockDagStorageTestFixture, BlockUtil, HashSetCasperTestNode}
+import coop.rchain.casper.protocol._
+import coop.rchain.casper.util.{BondingUtil, ProtoUtil}
+import coop.rchain.casper.util.ProtoUtil.{signBlock, toJustification}
+import coop.rchain.casper.util.rholang.RuntimeManager
+import coop.rchain.casper.util.rholang.InterpreterUtil.mkTerm
+import coop.rchain.catscontrib.TaskContrib.TaskOps
+import coop.rchain.comm.rp.ProtocolHelper.packet
+import coop.rchain.comm.{transport, CommError, TimeOut}
+import coop.rchain.crypto.{PrivateKey, PublicKey}
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.{Blake2b256, Keccak256}
+import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
+import coop.rchain.p2p.EffectsTestInstances.LogicalTime
+import coop.rchain.rholang.interpreter.{accounting, Runtime}
+import coop.rchain.models.{Expr, Par}
+import coop.rchain.shared.StoreType
+import coop.rchain.shared.PathOps.RichPath
+import coop.rchain.catscontrib._
+import coop.rchain.catscontrib.Catscontrib._
+import coop.rchain.catscontrib.eitherT._
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{Assertion, FlatSpec, Inspectors, Matchers}
+import coop.rchain.casper.scalatestcontrib._
+import coop.rchain.casper.util.comm.TestNetwork
+import coop.rchain.catscontrib.ski.kp2
+import coop.rchain.comm.rp.Connect.Connections
+import coop.rchain.metrics
+import coop.rchain.metrics.Metrics
+import coop.rchain.shared.Log
+import org.scalatest
+
+import scala.collection.immutable
+import scala.util.Random
+import scala.concurrent.duration._
+
+class MultiParentCasperRemainingTest extends FlatSpec with Matchers with Inspectors {
```

Comment:
I think with the remainder of tests this can be called "MPCRholangTest" or "MPCContentTest" - something that implies that we are doing a semi-integration tests involving both casper and the vm/rholang state